pkg/management/component_test.go

```diff
@@ -1699,6 +1699,57 @@ func TestSendReminderEmail(t *testing.T) {
 	}
 }
 
+func TestResetSmsCounter(t *testing.T) {
+	var mockCtrl = gomock.NewController(t)
+	defer mockCtrl.Finish()
+	var mockKeycloakClient = mock.NewKeycloakClient(mockCtrl)
+	var mockEventDBModule = mock.NewEventDBModule(mockCtrl)
+	var mockConfigurationDBModule = mock.NewConfigurationDBModule(mockCtrl)
+	var mockLogger = log.NewNopLogger()
+
+	var managementComponent = NewComponent(mockKeycloakClient, mockEventDBModule, mockConfigurationDBModule, mockLogger)
+
+	var accessToken = "TOKEN=="
+	var realmName = "master"
+	var userID = "1245-7854-8963"
+
+	// Reset SMS counter
+	{
+
+		mockKeycloakClient.EXPECT().GetUser(accessToken, realmName, userID).Return(kc.UserRepresentation{}, nil).Times(1)
+		mockKeycloakClient.EXPECT().UpdateUser(accessToken, realmName, userID, kc.UserRepresentation{}).Return(nil).Times(1)
+
+		var ctx = context.WithValue(context.Background(), cs.CtContextAccessToken, accessToken)
+
+		err := managementComponent.ResetSmsCounter(ctx, "master", userID)
+
+		assert.Nil(t, err)
+	}
+
+	// Error at GetUser
+	{
+
+		mockKeycloakClient.EXPECT().GetUser(accessToken, realmName, userID).Return(kc.UserRepresentation{}, fmt.Errorf("error")).Times(1)
```

Comment:
You also can write a test when m["smsSent"] is already set (with any value)

Reply:
Done